### ISSUE TITLE
Make: clang: error: argument unused during compilation: '-MD'

### ISSUE THREAD

**Issue by erikbylow-vol** [oneapi-src/oneTBB#470]
I have Macbook `BigSur 11.4` and `Xcode 12.5.1`

I have dowloaded `oneTBB-2021-2.0` and I want to build that from source.

I have run in `build` `cmake ..` and `make`.

Output from `cmake ..`:

```
cmake ..
-- The CXX compiler identification is AppleClang 12.0.5.12050022
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- CMAKE_BUILD_TYPE is not specified. Using default: RelWithDebInfo
-- Looking for C++ include pthread.h
-- Looking for C++ include pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- TBBBind build target is disabled due to unsupported environment
-- The C compiler identification is AppleClang 12.0.5.12050022
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/name/Downloads/oneTBB-2021.2.0/build
```
When running `make` I get immediately: 

```
[  0%] Building CXX object src/tbb/CMakeFiles/tbb.dir/allocator.cpp.o
clang: error: argument unused during compilation: '-MD' [-Werror,-Wunused-command-line-argument]
make[2]: *** [src/tbb/CMakeFiles/tbb.dir/allocator.cpp.o] Error 1
make[1]: *** [src/tbb/CMakeFiles/tbb.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by AlexVeprev** [comment]
Hi @erikbylow-vol! This issue looks very similar to https://github.com/oneapi-src/oneTBB/issues/364 that has been addressed in v2021.3.0 (master branch contains the fix as well).
As workaround for v2021.2.0 you can try to add `-DTBB_STRICT=OFF` to `cmake` command.

**Comment by erikbylow-vol** [comment]
Thanks, I think I cannot use any newer versions of TBB due to another library. (That is the reason to why I want to build from source in first place). I'll try that flag!
